app/src/main/java/org/fossasia/openevent/general/about/AboutEventFragment.kt

```diff
@@ -75,26 +73,24 @@ class AboutEventFragment : Fragment(), AppBarLayout.OnOffsetChangedListener {
         rootView.aboutEventContent.text = event.description?.stripHtml()
         val startsAt = EventUtils.getEventDateTime(event.startsAt, event.timezone)
         val endsAt = EventUtils.getEventDateTime(event.endsAt, event.timezone)
+
         rootView.aboutEventDetails.text = EventUtils.getFormattedDateTimeRangeBulleted(startsAt, endsAt)
         rootView.eventName.text = event.name
-    }
-
-    override fun onOffsetChanged(appBarLayout: AppBarLayout, verticalOffset: Int) {
-        val maxScroll = appBarLayout.totalScrollRange
-        val percentage = Math.abs(verticalOffset).toFloat() / maxScroll.toFloat()
+        Picasso.get()
+            .load(event.originalImageUrl)
+            .placeholder(R.drawable.header)
+            .into(rootView.aboutEventImage)
 
-        if (percentage == 1f && isHideToolbarView) {
-            // Collapsed
-            rootView.detailsHeader.visibility = View.GONE
-            rootView.aboutEventCollapsingLayout.title = title
-            isHideToolbarView = !isHideToolbarView
-        }
-        if (percentage < 1f && !isHideToolbarView) {
-            // Not Collapsed
-            rootView.detailsHeader.visibility = View.VISIBLE
-            rootView.aboutEventCollapsingLayout.title = " "
-            isHideToolbarView = !isHideToolbarView
-        }
+        rootView.appBar.addOnOffsetChangedListener(AppBarLayout.OnOffsetChangedListener { appBarLayout, offset ->
+            val thisActivity = activity
+            if (Math.abs(offset) == appBarLayout.getTotalScrollRange()) {
+                rootView.detailsHeader.isVisible = false
+                if (thisActivity is AppCompatActivity) thisActivity.supportActionBar?.title = event.name
+            } else {
+                rootView.detailsHeader.isVisible = true
+                if (thisActivity is AppCompatActivity) thisActivity.supportActionBar?.title = ""
```

Comment:
Just cast one time, when you define the value for `thisActivity`.

Reply:
Updates?

Reply:
oh yeah, I forgot to do that, fixing now

Reply:
@liveHarshit I have fixed it